pkg/adaptor/cloud/libvirt/libvirt.go

```diff
@@ -9,56 +9,22 @@ import (
 	"context"
 	"encoding/xml"
 	"fmt"
-	"log"
 	"net/netip"
-	"os"
-	"os/exec"
 	"strconv"
 	"time"
 
 	libvirt "libvirt.org/go/libvirt"
 	libvirtxml "libvirt.org/go/libvirtxml"
 )
 
-func createCloudInitISO(v *vmConfig, libvirtClient *libvirtClient) string {
-	logger.Printf("Create cloudInit iso\n")
-	cloudInitIso := libvirtClient.dataDir + "/" + v.name + "-cloudinit.iso"
+// createCloudInitISO creates an ISO file with a userdata and a metadata file. The ISO image will be created in-memory since it is small
+func createCloudInitISO(v *vmConfig) ([]byte, error) {
```

Comment:
All functions inside this file are libvirt related. If your proposal is changing the approach here, in terms of organization, maybe it makes sense to remove it from this file? Have you considered that?

Reply:
I wanted to keep the changes to existing code as minimal as possible.